fast64_internal/oot/data/oot_object_data.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from os import path
+from ...utility import PluginError
+from .oot_getters import getXMLRoot
+from .oot_data import OoT_BaseElement
+
+# Note: "object" in this context refers to an OoT Object file (like ``gameplay_keep``)
+
+
+@dataclass
+class OoT_ObjectElement(OoT_BaseElement):
+    index: int
+
+
+class OoT_ObjectData:
+    """Everything related to OoT objects"""
+
+    def __init__(self):
+        # general object list
+        self.objectList: list[OoT_ObjectElement] = []
+
+        # list of tuples used by Blender's enum properties
+        self.ootEnumObjectKey = [("Custom", "Custom Object", "Custom")]
+        self.ootEnumObjectIDLegacy = []  # for old blends
+
+        # Path to the ``ObjectList.xml`` file
+        objectXML = path.dirname(path.abspath(__file__)) + "/xml/ObjectList.xml"
+        objectRoot = getXMLRoot(objectXML)
+
+        for obj in objectRoot.iterfind("Object"):
+            objName = f"{obj.attrib['Name']} - {obj.attrib['ID'].removeprefix('OBJECT_')}"
+            self.objectList.append(
+                OoT_ObjectElement(obj.attrib["ID"], obj.attrib["Key"], objName, int(obj.attrib["Index"]))
+            )
+
+        self.objectsByID = {obj.id: obj for obj in self.objectList}
+        self.objectsByKey = {obj.key: obj for obj in self.objectList}
+        for obj in self.objectList:
+            self.ootEnumObjectKey.append((obj.key, obj.name, obj.id))
+
+        # create the legacy object list
+        lastIndex = self.objectsByKey["obj_timeblock"].index
+        self.ootEnumObjectIDLegacy = [None] * lastIndex
+        self.ootEnumObjectIDLegacy.insert(0, ("Custom", "Custom Object", "Custom"))
+        for obj in self.objectList:
+            if obj.index < lastIndex + 1:
+                self.ootEnumObjectIDLegacy[obj.index] = (obj.id, obj.name, obj.id)
+
+        # validate the legacy list, if there's any None element then something's wrong
+        if None in self.ootEnumObjectIDLegacy:
+            raise PluginError("ERROR: Legacy Object List doesn't match!")
```

Comment:
this check makes sense since we want the previous and hardcoded object list, so yea I think it should stay

Reply:
we only want it in the case the user has old blends, but it'd prevent fast64 from enabling even if they were starting from scratch and just deleted an object in the xml that they also removed in their repo

again this can be addressed later if needed though, just something to be aware of